feat: add `blas/base/wasm/scnrm2`

ShabiShett07 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

none

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds the package `blas/base/wasm/scnrm2`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   none 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

ShabiShett07 committed:
feat: add blas/base/wasm/scnrm2

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

ShabiShett07 commented:
/stdlbi update-copyright-years

ShabiShett07 committed:
remove: unused variables

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
fix: benchmark options

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/scnrm2/docs/repl.txt:
```diff
@@ -0,0 +1,500 @@
+    ~3.16
+
+    // Using offset parameter:
+    > var x = new {{alias:@stdlib/array/complex64}}( [ 3.0, 2.0, 3.0, 0.0 ] );
```
```suggestion
    > x = new {{alias:@stdlib/array/complex64}}( [ 3.0, 2.0, 3.0, 0.0 ] );
```

kgryte committed:
docs: avoid redeclaring variable

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/scnrm2/benchmark/benchmark.module.main.js:
```diff
@@ -0,0 +1,138 @@
+		// Initialize the module:
+		mod.initializeSync(); // eslint-disable-line node/no-sync
+
+		// Reallocate the underlying memory to allow storing one vector and a complex number:
+		nb = bytesPerElement( 'complex64' );
+		mod.realloc( (N*nb)+nb );
```
```suggestion
		// Reallocate the underlying memory to allow storing one vector:
		nb = bytesPerElement( 'complex64' );
		mod.realloc( N*nb );
```

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/scnrm2/benchmark/benchmark.module.ndarray.js:
```diff
@@ -0,0 +1,138 @@
+		// Initialize the module:
+		mod.initializeSync(); // eslint-disable-line node/no-sync
+
+		// Reallocate the underlying memory to allow storing one vector and a complex number:
+		nb = bytesPerElement( 'complex64' );
+		mod.realloc( (N*nb)+nb );
```
```suggestion
		// Reallocate the underlying memory to allow storing one vector:
		nb = bytesPerElement( 'complex64' );
		mod.realloc( N*nb );
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/scnrm2/test/test.main.js:
```diff
@@ -0,0 +1,170 @@
+		-2.0,
+		-2.0,
+		2.0,  // 2
+		3.0  // 2
```
```suggestion
		3.0   // 2
```

kgryte committed:
style: fix alignment

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/scnrm2/test/test.module.ndarray.js:
```diff
@@ -0,0 +1,232 @@
+		-2.0,
+		-2.0,
+		2.0,  // 2
+		3.0  // 2
```
```suggestion
		3.0   // 2
```

kgryte committed:
style: fix alignment

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/scnrm2/test/test.main.js:
```diff
@@ -0,0 +1,170 @@
+// FUNCTIONS //
+
+/**
+* Tests for element-wise approximate equality.
```
@ShabiShett07 Why are we testing for element-wise equality? The value returns a scalar, not an array. This same comment applies to all the tests. I am pretty sure these tests are all broken due to incorrect arguments. E.g., `expected.length` is undefined, as a number has no `length` property; hence, the loop never runs.

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/scnrm2/test/test.module.main.js:
```diff
@@ -0,0 +1,197 @@
+* @param {Collection} expected - expected values
+* @param {number} rtol - relative tolerance
+*/
+function isApprox( t, actual, expected, rtol ) {
```
Same comment.

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/scnrm2/test/test.module.ndarray.js:
```diff
@@ -0,0 +1,232 @@
+* @param {Collection} expected - expected values
+* @param {number} rtol - relative tolerance
+*/
+function isApprox( t, actual, expected, rtol ) {
```
Same comment.

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/scnrm2/test/test.ndarray.js:
```diff
@@ -0,0 +1,168 @@
+* @param {Collection} expected - expected values
+* @param {number} rtol - relative tolerance
+*/
+function isApprox( t, actual, expected, rtol ) {
```
Same comment.

kgryte reviewed (changes requested):
Other than the broken tests, this PR is looking good.

ShabiShett07 committed:
chore: fix test cases

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: fix test cases

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 commented:
@kgryte, I have fixed the broken test cases and also added more precise values with manual calculation. Also, I need your approval to make the same changes with other packages too.

kgryte commented on lib/node_modules/@stdlib/blas/base/wasm/scnrm2/test/test.ndarray.js:
```diff
@@ -0,0 +1,166 @@
+		-2.0,
+		-2.0,
+		2.0,  // 2
+		3.0  // 2
```
```suggestion
		3.0   // 2
```

kgryte committed:
style: align comment

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM

kgryte commented:
@ShabiShett07 Quick check: how did you calculate the expected values in the tests? Did you perform hand calculation?

ShabiShett07 commented:
> @ShabiShett07 Quick check: how did you calculate the expected values in the tests? Did you perform hand calculation?

Yes, squared all the input values made the sum and took the square root and cross checked with my calculator
